### PR TITLE
Unbreak build on FreeBSD by adjusting dependencies

### PR DESCRIPTION
**Pull request by jbeich**
CMake (unlike Meson) doesn't add `-Wl,--as-needed` by default, so a bit more verbose.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -174,7 +180,7 @@ if(NO_XWAYLAND)
     add_compile_definitions(NO_XWAYLAND)
 else()
     message(STATUS "XWAYLAND Enabled (NO_XWAYLAND not defined) checking deps...")
-    pkg_check_modules(xdeps REQUIRED IMPORTED_TARGET xcb xwayland xcb-util xcb-render xcb-image xcb-xfixes xcb-icccm xcb-composite xcb-res xcb-ewmh)
+    pkg_check_modules(xdeps REQUIRED IMPORTED_TARGET xcb xwayland xcb-util xcb-render xcb-xfixes xcb-icccm xcb-composite xcb-res xcb-ewmh)
```
Comment (vaxerski) [inline CMakeLists.txt:183]: why is xcb-image removed?

File: CMakeLists.txt [hunk 2]
```diff
@@ -164,6 +164,12 @@ if(HAVE_LIBEXECINFO)
     target_link_libraries(Hyprland execinfo)
 endif()
 
+check_include_file("sys/timerfd.h" HAS_TIMERFD)
+pkg_check_modules(epoll IMPORTED_TARGET epoll-shim)
+if(NOT HAS_TIMERFD AND epoll_FOUND)
```
Comment (vaxerski) [inline CMakeLists.txt:169]: but Hyprland hard-includes `<sys/timerfd.h>`, will epoll magically add that header?

Comment (jbeich) [reply CMakeLists.txt:169]: Yes:
```
$ pkg info -xl epoll-shim | fgrep timerfd
        /usr/local/include/libepoll-shim/sys/timerfd.h
$ pkg-config --cflags epoll-shim
-I/usr/local/include/libepoll-shim
```

Comment (vaxerski) [reply CMakeLists.txt:169]: cool

File: CMakeLists.txt (hunk 1 above)
Comment (jbeich) [reply CMakeLists.txt:183]: As explained in the commit message, it hasn't been used by wlroots for ~6 years.

Comment (vaxerski) [reply CMakeLists.txt:183]: but wlroots wouldn't compile for me without it.

Comment (jbeich) [reply CMakeLists.txt:183]: Can you quote the error message? CI is green here.

Comment (vaxerski) [reply CMakeLists.txt:183]: hm, odd. When I was making the original switch there were some messages referring to missing xcb-image symbols, but you are right, removing it works.

**Review by vaxerski — APPROVED** [review]

**Pull request merged**